### PR TITLE
[BEAM-9320] Add AlwaysFetched annotation

### PR DESCRIPTION
**Pull request by reuvenlax**

### CODE REVIEW

**Comment by reuvenlax** [comment]
R: @dpmills

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFn.java [hunk 1]
```diff
@@ -397,6 +397,47 @@ public Duration getAllowedTimestampSkew() {
...
+  /////////////////////////////////////////////////////////////////////////////
+
+  /**
+   * Annotation for declaring that a state parameter is always fetched.
+   *
+   * <p>A DoFn might not fetch a state value on every element, and for that reason runners may
+   * choose to defer fetching state until read() is called. Annotating a state argument with this
+   * parameter provides a hint to the runner that the state is always fetched. This may cause the
+   * runner to prefetch all the state before calling the processElement or processTimer method,
+   * improving performance. See the following code for an example:
```
Comment (dpmills) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFn.java]: Add a comment that the annotation only works for ReadableState

**Comment by reuvenlax** [comment]
run dataflow validatesrunner

**Comment by reuvenlax** [comment]
run flink validatesrunner

**Comment by reuvenlax** [comment]
Run Java PreCommit

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFn.java (hunk 1 above)
Comment (reuvenlax) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFn.java]: done

**Comment by reuvenlax** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Comment by reuvenlax** [comment]
run dataflow validatesrunner

**Comment by reuvenlax** [comment]
run flink validatesrunner

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Pull request merged**